misc pkgs: Fewer cross attrs

Ericson2314 opened the pull request:
###### Motivation for this change

Progress towards #33302

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Ericson2314 committed:
SDL, SDL2: Remove crossAttrs

Cross drvs are now the default for precisely this reason.

Ericson2314 committed:
freetype: Remove crossAttrs

These sorts of C compiler confusions aren't really a problem anymore.
Everything is on the path with the right environment variables.

Ericson2314 committed:
libdrm: Remove crossAttrs

No hashes should be changed.

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40291&attempt_id=a14a6a25-592e-4c2b-9623-fffb4bf240dd)

Attempted: SDL, SDL2, freetype

The following builds were skipped because they don't evaluate on x86_64-darwin: libdrm

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/4v2y8wgf6dhdwbi9p6kjnipxkbjbj4cm-SDL2-2.0.8/lib/libSDL2-2.0.0.dylib: fixing dylib
/nix/store/4v2y8wgf6dhdwbi9p6kjnipxkbjbj4cm-SDL2-2.0.8/lib/libSDL2-2.0.0.dylib: fixing dylib
/nix/store/4v2y8wgf6dhdwbi9p6kjnipxkbjbj4cm-SDL2-2.0.8/lib/libSDL2-2.0.0.dylib: fixing dylib
strip is /nix/store/kdff2gim6417493yha769kh00n63lnrw-cctools-binutils-darwin/bin/strip
stripping (with command strip and flags -S) in /nix/store/864wsjyy02wy04pzi86qjmm5ngac9v6g-SDL2-2.0.8-dev/lib  /nix/store/864wsjyy02wy04pzi86qjmm5ngac9v6g-SDL2-2.0.8-dev/bin
patching script interpreter paths in /nix/store/864wsjyy02wy04pzi86qjmm5ngac9v6g-SDL2-2.0.8-dev
/nix/store/864wsjyy02wy04pzi86qjmm5ngac9v6g-SDL2-2.0.8-dev/bin/sdl2-config: interpreter directive changed from "/bin/sh" to "/nix/store/r8bx3qf1bpncb14i9gzma4vr089pc3pv-bash-4.4-p19/bin/sh"
/nix/store/8r9y5p25gvr0hpplf1ql8ckmb9zdxij5-SDL-1.2.15
/nix/store/4v2y8wgf6dhdwbi9p6kjnipxkbjbj4cm-SDL2-2.0.8
/nix/store/7xzr0r22z42c931ax7nqg2vvmclyl6ji-freetype-2.9
```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40291&attempt_id=6cb1d02c-e928-4b99-b6c2-ae82b1acc2eb)

Attempted: SDL, SDL2, freetype, libdrm

<details><summary>Partial log (click to expand)</summary><p>

```

Alternatively, you may set the environment variables I915_CFLAGS
and I915_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.
builder for '/nix/store/r6wgs59f06a6imh60fgwiafyqch8p3hz-mesa-noglu-17.3.8.drv' failed with exit code 1
cannot build derivation '/nix/store/wy1hg60im38zxgrq14grvq48c7kb229q-libGL-1.0.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/7dl0lw9a2gkblcfvx5rqyvybwcjib16c-SDL2-2.0.8.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/cfaw35pm71hx63xx4i6g17w5rxzsx9sf-glu-9.0.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/bl7b8nys6bvmgm1rg5ihcl8ggayzgzp2-SDL-1.2.15.drv': 2 dependencies couldn't be built
error: build of '/nix/store/7dl0lw9a2gkblcfvx5rqyvybwcjib16c-SDL2-2.0.8.drv', '/nix/store/bl7b8nys6bvmgm1rg5ihcl8ggayzgzp2-SDL-1.2.15.drv' failed
```
</p></details>

Ericson2314 committed:
libpng: Remove stale crossAttrs

Random ancient stuff for Darwin cross that seems not worth it.

Ericson2314 committed:
libssh2: Remove crossAttrs

 - --with-<lib> stuff isn't needed with new default

 - `mingw_w64` probably also isn't needed, but kept with better code
   just in case.

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40291&attempt_id=dee28003-d4eb-44e8-97d0-f5db0dd94ac0)

Attempted: SDL, SDL2, freetype

The following builds were skipped because they don't evaluate on x86_64-darwin: libdrm

<details><summary>Partial log (click to expand)</summary><p>

```
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


/nix/store/8r9y5p25gvr0hpplf1ql8ckmb9zdxij5-SDL-1.2.15
/nix/store/4v2y8wgf6dhdwbi9p6kjnipxkbjbj4cm-SDL2-2.0.8
/nix/store/7xzr0r22z42c931ax7nqg2vvmclyl6ji-freetype-2.9
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40291&attempt_id=442c66a4-6b5b-46d2-ac01-4b10cec910c6)

Attempted: SDL, SDL2, freetype, libdrm

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking RPATHs of ELF executables and libraries in /nix/store/mcyyvjj9zjaf7r55g5xs39q41sdhzf17-SDL2-2.0.8-dev
strip is /nix/store/j75dgadrff2d1fyc4fczmcgqkid2imdx-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/mcyyvjj9zjaf7r55g5xs39q41sdhzf17-SDL2-2.0.8-dev/lib  /nix/store/mcyyvjj9zjaf7r55g5xs39q41sdhzf17-SDL2-2.0.8-dev/bin
patching script interpreter paths in /nix/store/mcyyvjj9zjaf7r55g5xs39q41sdhzf17-SDL2-2.0.8-dev
/nix/store/mcyyvjj9zjaf7r55g5xs39q41sdhzf17-SDL2-2.0.8-dev/bin/sdl2-config: interpreter directive changed from "/bin/sh" to "/nix/store/xn5gv3lpfy91yvfy9b0i7klfcxh9xskz-bash-4.4-p19/bin/sh"
checking for references to /build in /nix/store/mcyyvjj9zjaf7r55g5xs39q41sdhzf17-SDL2-2.0.8-dev...
/nix/store/8hr6xww9cjrc7lm0k4x0cl44x6qzvvcn-SDL-1.2.15
/nix/store/25crnb1gvf03sf40c51hd83ssiv1aa11-SDL2-2.0.8
/nix/store/3192bm9pzlzj1f4nqf16zvaqr73k2fhc-freetype-2.9
/nix/store/y7xyva8ra4r1frpkz42i7xxacnicyxc8-libdrm-2.4.91
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40291&attempt_id=bcca4028-3211-4b9f-8548-99899e0da0ad)

Attempted: SDL, SDL2, freetype, libdrm

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking RPATHs of ELF executables and libraries in /nix/store/296v36xf4hvq9ffnalwljw7ya23m8zhp-SDL2-2.0.8-dev
strip is /nix/store/gp7fylxwn18b7pl2c18ks89hsiaxyfvf-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/296v36xf4hvq9ffnalwljw7ya23m8zhp-SDL2-2.0.8-dev/lib  /nix/store/296v36xf4hvq9ffnalwljw7ya23m8zhp-SDL2-2.0.8-dev/bin
patching script interpreter paths in /nix/store/296v36xf4hvq9ffnalwljw7ya23m8zhp-SDL2-2.0.8-dev
/nix/store/296v36xf4hvq9ffnalwljw7ya23m8zhp-SDL2-2.0.8-dev/bin/sdl2-config: interpreter directive changed from "/bin/sh" to "/nix/store/a4qslf0yfs44mhnbagarn232sjnckhy0-bash-4.4-p19/bin/sh"
checking for references to /build in /nix/store/296v36xf4hvq9ffnalwljw7ya23m8zhp-SDL2-2.0.8-dev...
/nix/store/g9f18skhgv6gw2nfx811wlr5smyv3rs0-SDL-1.2.15
/nix/store/4qhsr779zzvrq28wv150sxj2slxwdg99-SDL2-2.0.8
/nix/store/ia9b9879xmipfb4xssi8r75ssdwyf2a3-freetype-2.9
/nix/store/2jj6pirxhm1f2f66712zdm42lj9ciwam-libdrm-2.4.91
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40291&attempt_id=54af85b7-4d62-43b8-8ac8-8ed5a9fb9f04)

Attempted: SDL, SDL2, freetype, libdrm, libpng, libssh2

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/8hr6xww9cjrc7lm0k4x0cl44x6qzvvcn-SDL-1.2.15
/nix/store/580y11zq14xag6ybrvzcbr7k93naman4-SDL2-2.0.8
/nix/store/3192bm9pzlzj1f4nqf16zvaqr73k2fhc-freetype-2.9
/nix/store/y7xyva8ra4r1frpkz42i7xxacnicyxc8-libdrm-2.4.91
/nix/store/1n47iy8z04x07ahq8phzdp1bvmqwfmmc-libpng-apng-1.6.34
/nix/store/8kanfjdi2yh998xbinc1m8ywysnf41k6-libssh2-1.8.0
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40291&attempt_id=e0594736-2b94-482b-875d-5c404f8bf7c7)

Attempted: SDL, SDL2, freetype, libdrm, libpng, libssh2

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/g9f18skhgv6gw2nfx811wlr5smyv3rs0-SDL-1.2.15
/nix/store/57xsz4ddqhcs5hcq084p68cjd19mabkz-SDL2-2.0.8
/nix/store/ia9b9879xmipfb4xssi8r75ssdwyf2a3-freetype-2.9
/nix/store/2jj6pirxhm1f2f66712zdm42lj9ciwam-libdrm-2.4.91
/nix/store/87s3ak35aq4svj4165lpq2qi5gzs2snm-libpng-apng-1.6.34
/nix/store/7f538fs1b88fssc7qmq8hkc7pbc5cbps-libssh2-1.8.0
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40291&attempt_id=bc7fd291-9af5-4e7f-994e-ac47e22528b9)

Attempted: SDL, SDL2, freetype, libpng, libssh2

The following builds were skipped because they don't evaluate on x86_64-darwin: libdrm

<details><summary>Partial log (click to expand)</summary><p>

```
b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


/nix/store/8r9y5p25gvr0hpplf1ql8ckmb9zdxij5-SDL-1.2.15
/nix/store/3f9qv287i405js11zgbdwgcy3c5a8mrd-SDL2-2.0.8
/nix/store/7xzr0r22z42c931ax7nqg2vvmclyl6ji-freetype-2.9
/nix/store/s7ra5sq6na42fj93xw2bay80dhpf68dk-libpng-apng-1.6.34
/nix/store/mwdkj4cakd5dg62ai11g4dfgswk6q0xx-libssh2-1.8.0
```
</p></details>
